content/articles/build-autocorrect-feauture-using-nlp-with-python/index.md

```diff
@@ -0,0 +1,285 @@
+```python
+pip install pattern
+pip install pyspellchecker
+pip install autocorrect
+pip install textblob
+pip install textdistance
+```
+#### Reading text file(dictionary)
+We import all the necessary libraries and packages then read the text file containing our vocabulary dictionary with the code below.
+```python
+# Step 1: Data Preprocessing
+import re  # regular expression
+from collections import Counter
+import numpy as np
+import pandas as pd
+
+
+# Implement the function process_data which
+# 1) Reads in a corpus
+# #2) Changes everything to lowercase
+# 3) Returns a list of words.
+
+
+words = []
+with open('sample.txt','r',encoding="utf8") as f:
+    file_name_data = f.read()
+    file_name_data = file_name_data.lower()
+    words = re.findall('\w+', file_name_data)
+
+
+vocab = set(words)
+print(f"The first ten words in the text are: \n{words[0:10]}")
+print(f"There are {len(vocab)} unique words in the vocabulary.")
+```
+The following output is displayed in the terminal showing  number of unique words and the top 10 words in the dictionary. 
+
+```bash
+The first ten words in the text are: 
+['a', 'ability', 'able', 'about', 'above', 'accept', 'according', 'account', 'across', 'act']
+There are 1001 unique words in the vocabulary.
+```
+#### Frequency and Probability of Dictionary Words 
+We use the `counter function` to find the frequency of the words by applying the code below.
+```python
+# a get_count function that returns a dictionary of word vs frequency
+def get_count(words):
+    word_count_dict = {}
+    for word in words:
+        if word in word_count_dict:
+            word_count_dict[word] += 1
+        else:
+            word_count_dict[word] = 1
+    return word_count_dict
+word_count_dict = get_count(words)
+print(f"There are {len(word_count_dict)} key values pairs")
+```
+The output displayed is as shown below with 1001 key values pairs: 
+
+```bash
+The first ten words in the text are: 
+['a', 'ability', 'able', 'about', 'above', 'accept', 'according', 'account', 'across', 'act']
+There are 1001 unique words in the vocabulary.
+There are 1001 key values pairs
+```
+We compute the probability that each word will appear if randomly selected from the corpus of words using the `get_probs function` as shown in the code.
+```python
+
+# Compute the probability that each word will appear if randomly selected from the corpus of words.
+# implement get_probs function
+
+def get_probs(word_count_dict):
+    probs = {}
+    m = sum(word_count_dict.values())
+    for key in word_count_dict.keys():
+        probs[key] = word_count_dict[key] / m
+    return probs
+```
+#### Implement 4 edit word functions
+Each of the 4 edit functions is implemented as shown below each performing different tasks as early illustrated. We can also combine the functions to be able to find a list of all the strings that are n edits to perform.
```

Comment:
Change to " Each of the four edit functions is implemented below, each performing different tasks as illustrated earlier. We can also combine the functions with finding a list of all the strings that are n edited to execute."